### PR TITLE
[Merged by Bors] - feat(ConditionallyCompleteLattice/Finset): ciSup_mem on list/multiset/finset

### PR DESCRIPTION
**Pull request by pechersky**
As well as toDual_csSup
And simpler lemmas when on ConditionallyCompleteLinearOrderBot

---

- [x] depends on: #15271

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:84]:
```suggestion
      exists_exists_and_eq_and, forall_exists_index, forall_apply_eq_imp_iff]
```

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:91]:
```suggestion
      exists_exists_eq_and, forall_exists_index, and_imp, forall_apply_eq_imp_iff₂]
```

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:137]: Squeeze

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:161]: Squeeze

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:286]: What's BigOperators about this?

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (pechersky) [reply Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:286]: That is operates over List and Multiset. Do you have a better section name?

Comment (pechersky) [reply Mathlib/Order/ConditionallyCompleteLattice/Finset.lean]: renamed section to `ListMultiset`

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by urkud** [comment]
I think that some lemmas in this PR can be generalized. Please don't merge it yet. I'll write more in ~1h.

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean
Comment (urkud) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:301]: This lemma should assume `[SupSet α]`, same for the next one. Do we use `ciSup` for lemmas like this? I thought that this is for conditionally complete lattice versions of lemmas that have `iSup` versions with fewer assumptions.

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (urkud) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:50]: This is true for any set `s`, because this is basically `congr_arg sSup (image_id s)`

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean
Comment (pechersky) [reply Mathlib/Order/ConditionallyCompleteLattice/Basic.lean]: Done, and made sure to generalize the existing lemmas out of `CompleteLattice` to `SupSet`.

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (pechersky) [reply Mathlib/Order/ConditionallyCompleteLattice/Finset.lean]: Done, and made sure to generalize the existing lemmas out of `CompleteLattice` to `SupSet`.

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Vierkantor) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:159]: `simpa` (unlike `rwa`) accepts a full term so we can simplify to:
```suggestion
  simpa using l.toFinset.ciSup_mem_image f (by simpa using h)
```

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Vierkantor) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:166]:
```suggestion
  simpa using l.toFinset.ciSup_mem_image f (by simpa using h)
```

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Vierkantor) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:173]:
```suggestion
  simpa using s.toFinset.ciSup_mem_image f (by simpa using h)
```

File: Mathlib/Order/ConditionallyCompleteLattice/Finset.lean
Comment (Vierkantor) [inline Mathlib/Order/ConditionallyCompleteLattice/Finset.lean:180]:
```suggestion
  simpa using s.toFinset.ciInf_mem_image f (by simpa using h)
```

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors d+

**Pull request closed without merging**
